Forms/DrawForm/DrawForm.cs

```diff
@@ -57,15 +81,35 @@ private void MouseClickDrawForm(object sender, MouseEventArgs e)
                     , (int)VertexParameters.Width, (int)VertexParameters.Height, "Саси"
                     , vertexDraws.Count);
 
+
                 if (collisionVertex.IsDrawVertex(vertexDraw, vertexDraws))
                 {
                     vertexDraws.Add(vertexDraw);
 
                     Refresh();
                 }
+                else
+                {
+                    vertexClick.VertexRemember(ref startVertexId, ref endVertexId
+                        , e.X - (int)VertexParameters.Radius, e.Y - (int)VertexParameters.Radius
+                        , vertexDraws, (int)VertexParameters.Radius);
+
+                    if ((startVertexId != -1) && (endVertexId != -1) && (startVertexId != endVertexId))
+                    {
+                        EdgeDraw edgeDraw = new EdgeDraw(BrushColor.Black, 0, startVertexId, endVertexId);
+
+                        edgeDraws.Add(edgeDraw);
+                        Refresh();
+
+                        startVertexId = -1;
+                        endVertexId = -1;
+                    }
+
```

Comment:
Too complicated logic, separate class needed

Reply:
fuck you, dominik